PDBminer

```diff
@@ -1649,109 +1631,102 @@ def cleanup_all(structural_df,input_dataframe, path, output_format):
-            input_file = 'input_file.csv'
-            f.write("input file created based on hugo name and uniprot id, file name: input_file.csv\n")
-    
-        else:
-            f.write("ERROR: input file or input hugo name and uniprot id is missing. Exiting...\n")
+    if args.inputfile:
+        input_file = args.inputfile
+        logging.info("input file mode is choosen and input file has been identified.")
+
+        df = pd.read_csv(input_file, na_values=[], keep_default_na=False)
+        
+        uniprot_problems = uniprot_accession_number_type(df)
+        if uniprot_problems is not None:
+            logging.error(f"ERROR: The input file is not correctly formated. {uniprot_problems}. Exiting...")
             exit(1)
-            
-        f.write("Pipeline is starting\n")
         
-        run_list(input_file, path, args.cores, args.format)
+        df,problems = check_input_file(df)
+
+        if problems == []:
+            df.to_csv(input_file, index=False)
+
+        elif len(problems) > 0:
+            logging.error(f"ERROR: The input file is not correctly formated. {problems}. Exiting...")
+            exit(1)
+
+    elif args.uniprot_id:
+        uniprot_id = [args.uniprot_id]
+        if args.hugo_name:
+            hugo_name = [args.hugo_name]
+        else:  
+            hugo_name = retrieve_hugo_name(uniprot_id)
+        
+        df = pd.DataFrame({'hugo_name':hugo_name,'uniprot':uniprot_id})
+        
+        if args.uniprot_isoform:
+            isoform = [args.uniprot_isoform]
+            df['uniprot_isoform'] = isoform
+        else: df['uniprot_isoform'] = [1]
+        
+        if args.mutations:
+            mutations = [args.mutations]
+            df['mutations'] = mutations
+        else: df['mutations'] = ["NA"]
+        
+        if args.cluster_id:
+            cluster = [args.cluster_id]
+            df['cluster_id'] = cluster
         
-        finished = datetime.now() 
+        else: df['cluster_id'] = ["NA"]
+        df.to_csv(f'{path}/input_file.csv')
+        input_file = 'input_file.csv'
+        logging.info("input file created based on hugo name and uniprot id, file name: input_file.csv")
+
+    else:
+        logging.error("ERROR: input file or input hugo name and uniprot id is missing. Exiting...")
+        exit(1)
         
-        dirs = glob.glob("results/*/structures")
-        for directory in dirs:
-            shutil.rmtree(directory)
-        logs = glob.glob("results/*/log.txt")
-        for log in logs:
-            if os.path.isfile(log) and os.path.getsize(log) < 10:
-                os.remove(log)
-
-        f.write(f"time: {finished-start}\n")
-        f.write("PDBminer has finished.\n")
+    logging.info("Pipeline is starting")
+    
+    run_list(input_file, path, args.cores, args.format, args.peptide_length)
+    
+    finished = datetime.now() 
+    
+    dirs = glob.glob("results/*/structures")
+    for directory in dirs:
+        shutil.rmtree(directory)
```

Comment:
I would be safer if we could explicitly list the directories that we actually want to delete instead of glowing for `*` - removing directories in a not truly predictable fashion like this can have unforeseen consequences